components/shapediver/viewport/ViewportIcons.tsx

```diff
@@ -1,289 +1,183 @@
+				if (!namespace) {
+					return [];
+				}
+
+				const ids = state.sessionDependency[namespace];
+				return ids
+					.map((id) => state.parameterChanges[id])
+					.filter(Boolean);
+			},
+			[namespace],
+		),
 	);
 
-	const isFullscreenDisabled = !enableFullscreenBtn || isIPhone();
-
-	const {makeElementFullscreen, isFullScreenAvailable} =
-		useFullscreen(fullscreenId);
-
-	const cameras = enableCamerasBtn && viewport ? viewport.cameras : {};
-	const noCamerasAvailable = Object.keys(cameras).length === 0;
-
-	const [isCamerasMenuOpened, setIsCamerasMenuOpened] = useState(false);
-
-	const onCameraSelect = (cameraId: string) => {
-		if (!viewport) return;
+	const executing = useMemo(
+		() => parameterChanges.some((change) => change.executing),
+		[parameterChanges],
+	);
 
-		viewport.assignCamera(cameraId);
-	};
+	const hasPendingChanges = useMemo(
+		() =>
+			parameterChanges.length > 0 &&
+			parameterChanges.some((c) => Object.keys(c.values).length > 0),
+		[parameterChanges],
+	);
 
-	const cameraElements = Object.values(cameras).map((camera, i) => {
-		return (
-			<Menu.Item onClick={() => onCameraSelect(camera.id)} key={i}>
-				{firstLetterUppercase(camera.name || camera.id)}
-			</Menu.Item>
-		);
-	});
+	const buttonsDisabled = hasPendingChanges;
+	const isArEnabled = viewport ? viewport.enableAR : false;
 
 	return (
-		<Box style={style}>
-			{enableArBtn && isArEnabled && (
-				<TooltipWrapper label="View in AR">
-					<div>
-						<ActionIcon
-							onClick={onArClick}
-							disabled={isArLoading}
-							size={size}
-							variant={isViewableInAr ? variantDisabled : variant}
-							aria-label="View in AR"
-							style={iconStyle}
-						>
-							<Icon
-								type={IconTypeEnum.AugmentedReality}
-								color={isArLoading ? colorDisabled : color}
-								className={classes.viewportIcon}
-							/>
-						</ActionIcon>
-					</div>
-				</TooltipWrapper>
-			)}
-
-			{enableArBtn && (
-				<Modal
-					opened={isModalArOpened}
-					onClose={() => setIsModalArOpened(false)}
-					title="Scan the code"
-					centered
-				>
-					{isModalArError ? (
-						<Text c="red">{isModalArError}</Text>
-					) : (
-						<>
-							<Text>
-								Scan the QR code below using your mobile device
-								to see the model in AR. The code is compatible
-								with Android and iOS devices.
-							</Text>
-							<section className={classes.containerAr}>
-								{isArLoading ? (
-									<section className={classes.loaderAr}>
-										<Loader />
-									</section>
-								) : (
-									<img
-										src={arLink}
-										height="180px"
-										alt="ar_link"
-									/>
-								)}
-							</section>
-						</>
-					)}
-				</Modal>
-			)}
-
-			{enableZoomBtn && (
-				<TooltipWrapper label="Zoom extents">
-					<ActionIcon
-						onClick={zoomClickHandler}
-						size={size}
+		<ViewportOverlayWrapper position={position} offset={offset}>
+			<Paper style={style} shadow={shadow} py={py} px={px}>
```

Comment:
@LemToUp given that we are already refactoring the theme properties of this component (and potentially breaking existing themes), what about changing this to `{...viewportOverlayProps}` respectively `{...paperProps}`? This seems easier to maintain in the future and we don't need to refactor should we ever extend `ViewportOverlayWrapper` by further properties.